plone/app/linkintegrity/testing.py

```diff
@@ -73,7 +74,9 @@ def setUpContent(self):
                 create(portal, **type_data)
 
             create(portal, 'File', id='file1', title='File 1', file=GIF)
-            create(portal, 'Image', id='image1', title='Image 1', image=GIF)
+            portal.invokeFactory('Image', 'image1')
+            portal['image1'].image = NamedImage(
+                GIF, 'image/gif', u'sample.gif')
```

Comment:
The tests for p.a.linkintegrity are run both with Dexterity and Archetypes. `NamedImage` is only DX though. You'd have to create a wrapper-method similar to `create` that sets the image using in different methods for the respective ct-framework.